controllers/disruption_controller.go

```diff
@@ -652,12 +648,20 @@ func (r *DisruptionReconciler) selectTargets(instance *chaosv1beta1.Disruption)
 
 // getChaosPods returns chaos pods owned by the given instance and having the given labels
 func (r *DisruptionReconciler) getChaosPods(instance *chaosv1beta1.Disruption, ls labels.Set) ([]corev1.Pod, error) {
-	ownedPods := make([]corev1.Pod, 0)
 	pods := &corev1.PodList{}
 
+	if ls == nil {
+		ls = make(map[string]string)
+	}
+
+	// If these aren't valid labels, we will instead list ALL pods
+	ls[chaostypes.DisruptionNameLabel] = instance.Name
+	ls[chaostypes.DisruptionNamespaceLabel] = instance.Namespace
+
+	r.log.Infow("Getting chaos pods with this label", "labels", ls.String())
+
 	// list pods in the instance namespace and for the given target
 	listOptions := &client.ListOptions{
-		Namespace:     instance.Namespace,
```

Comment:
Shouldn't we keep the namespace here but with `r.InjectorServiceAccountNamespace` instead? So we still scope the listing to a specific namespace (better to not be too harmful with apiserver I think)

Reply:
Good point